packages/golden-layout/src/LayoutManager.ts

```diff
@@ -44,6 +50,14 @@ export type ComponentConstructor<
   new (container: ItemContainer<C>, state: unknown): unknown;
 };
 
+type ItemConfigSupported =
+  | ColumnItemConfig
+  | RowItemConfig
+  | StackItemConfig
+  | ComponentConfig;
+
+const SUPPORTED_ITEM_TYPES = ['column', 'row', 'stack', 'component'] as const;
```

Comment:
Docstrings for why these are the only supported types? Why isn't `ReactComponentConfig` supported for example? What is this "supported" by? We use `ReactComponentConfig` in a lot of places elsewhere... Maybe a better name would be something like `LayoutItemConfig` ?

Reply:
@bender I don't actually know the answer as to why we only support these. This was just based on the constructors that were in the previous code. Any ideas?

Reply:
I guess the `ReactComponentConfig` isn't listed due to it being converted out where this is used. I can improve that in the names / docs here to be more explicit, but not sure of any other reasons why other types aren't supported.

Reply:
Updated based on what I know. Not sure why DefaultItemConfig and RootItemConfig aren't included, but should at least clarify how things are used.